docker/Dockerfile

```diff
@@ -0,0 +1,104 @@
+# build stage for compiling odamex
+FROM ubuntu:20.04 AS build
+WORKDIR /build
+ENV DEBIAN_FRONTEND=noninteractive
+RUN true \
+    && apt-get update -qq \
+    && apt-get install -y --no-install-recommends tzdata\
+    && apt-get install -y -qq --no-install-recommends \
+        ca-certificates \
+        git \
+        gcc g++ \
+        cmake \
+        make \
+        deutex \
+        libssl-dev \
+        libsdl2-dev libsdl2-mixer-dev \
+        libpng-dev libpng++-dev \
```

Comment:
odasrv doesn't need a lot of these. SDL, SDL_mixer, libpng, and cURL are only used by the client.